src/sagemaker/jumpstart/enums.py

```diff
@@ -126,3 +128,88 @@ def from_suffixed_type(mime_type_with_suffix: str) -> "MIMEType":
+    def upper_camel_to_snake(upper_camel_case_string: str) -> str:
+        """Converts UpperCamelCaseString to snake_case_string."""
+        snake_case_string = re.sub("(.)([A-Z][a-z]+)", r"\1_\2", upper_camel_case_string)
+        return re.sub("([a-z0-9])([A-Z])", r"\1_\2", snake_case_string).lower()
+
+    @staticmethod
+    def snake_to_upper_camel(snake_case_string: str) -> str:
+        """Converts snake_case_string to UpperCamelCaseString."""
+        upper_camel_case_string = "".join(word.title() for word in snake_case_string.split("_"))
+        return upper_camel_case_string
+
+
+class ModelSpecKwargType(str, Enum):
+    """Enum class for types of kwargs for model hub content document and model specs."""
+
+    FIT = "fit_kwargs"
+    MODEL = "model_kwargs"
+    ESTIMATOR = "estimator_kwargs"
+    DEPLOY = "deploy_kwargs"
+
+    @classmethod
+    def arg_keys(cls) -> List[str]:
+        """Returns a list of kwargs keys that each type can have"""
+        return [member.value for member in cls]
+
+    @staticmethod
+    def get_model_spec_arg_keys(
+        arg_type: "ModelSpecKwargType",
+        naming_convention: NamingConventionType = NamingConventionType.DEFAULT,
+    ) -> List[str]:
+        """Returns a list of arg keys for a specific model spec arg type.
+
+        Args:
+            arg_type (ModelSpecKwargType): Type of the model spec's kwarg.
+            naming_convention (NamingConventionType): Type of naming convention to return.
+
+        Raises:
+            ValueError: If the naming convention is not valid.
+
+        """
+        arg_keys = []
+        if arg_type == ModelSpecKwargType.DEPLOY:
+            arg_keys = ["ModelDataDownloadTimeout", "ContainerStartupHealthCheckTimeout"]
+        elif arg_type == ModelSpecKwargType.ESTIMATOR:
+            arg_keys = [
+                "EncryptInterContainerTraffic",
+                "MaxRuntimeInSeconds",
+                "DisableOutputCompression",
+            ]
+        elif arg_type == ModelSpecKwargType.MODEL:
+            arg_keys = []
+        elif arg_type == ModelSpecKwargType.FIT:
+            arg_keys = []
+        if naming_convention == NamingConventionType.SNAKE_CASE:
+            return NamingConventionType.upper_camel_to_snake(arg_keys)
+        elif naming_convention == NamingConventionType.UPPER_CAMEL_CASE:
+            return arg_keys
+        else:
+            raise ValueError("Please provide a valid naming convention.")
+
+    @staticmethod
+    def get_model_spec_kwargs_from_hub_content_document(
+        arg_type: "ModelSpecKwargType",
+        hub_content_document: Dict[str, Any],
+        naming_convention: NamingConventionType = NamingConventionType.UPPER_CAMEL_CASE,
+    ) -> Dict[str, Any]:
+        kwargs = dict()
+        keys = ModelSpecKwargType.get_model_spec_arg_keys(
+            arg_type, naming_convention=naming_convention
+        )
+        for k in keys:
+            kwarg_value = hub_content_document.get(k, None)
```

Comment:
Do we need to worry about nested keys here?

Reply:
I believe not since kwarg dict depth is just one.